test/unit/puppetlabs/puppetserver/certificate_authority_test.clj

```diff
@@ -431,7 +444,14 @@
           (is (thrown-with-msg?
                IllegalStateException
                (re-pattern (str "Missing:\n" path))
-               (initialize! settings))))))))
+               (initialize! settings)))))))
+
+  (testing "The CA private key has its permissions properly reset"
+    (let [settings (testutils/ca-sandbox! cadir)]
+      (set-file-perms (:cakey settings) "rw-r--r--")
+      (logutils/with-test-logging
+        (initialize! settings)
+        (is (logged? #"/ca/ca_key.pem' was found to have the wrong permissions set as 'rw-r--r--'. This has been corrected to 'rw-r-----'."))))))
```

Comment:
Would seem good to check the file's permissions as well to see that they were reset, not just check for a log message.